### ISSUE TITLE
CLI metadata: allow autoinstall

### ISSUE THREAD

**Issue by josephjclark** [OpenFn/kit#192]
When you run the CLI metadata command:
```
openfn metadata -a salesforce -s tmp/state.json
```
The command will fail if the adaptor has not been previously  installed to the CLI's repo.

Adaptors can be autoinstalled in the CLI either by running a job/workflow, or by running `openfn repo install <adaptor>`

The work in this issue is to enable the metadataservice to autoinstall an adaptor. 

## Background

Metadata is used to drive "magic functions" in Lightning.

The idea is that each adaptor can provide a `metadata()` export which is responsible for loading a JSON-based description of the backend entities. In Salesforce for example that means logging into Salesforce and getting a description of all the sObjects on that deployment.

The metadata command is responsible for calling that metadata function with a credential and saving the result to disk.

At the time of writing, we only provide metadata for Salesforce and DHIS2.

## Implementation Notes

The `execute` service already has an autoinstall capability. I think you basically just need to import and call the autoinstall function from within the metadata service. 

Check out the implementation in `packages/cli/src/execute/handler.ts` for reference.

## Testing

Any solution must include unit tests. Unit tests should not run `npm install` or  fetch any external files, so you may have to work out a means of mocking out the autoinstall function (right now its imported from the runtime package, but perhaps you could pass a mock override - see the pattern in worker autoinstall at `packages/engine-multi/test/api/autoinstall.test.ts`)

We have a suite of metadata tests already in `integration-tests/cli/test/metadata.test.ts`. These use a local , pre-installed test adaptor to test the functionality.

I do not want to trigger an actual metadata run against a third party service in our unit tests. It would be acceptable to run autoinstall on the common adaptor. This should fail to generate metadata (because common has no metadata function), but it should do the autoinstall. So you could test that correct error is issued by the metadata command (something like `no metadata helper found`, as reported in `packages/cli/src/metadata/handler.ts`)

**Comment by SatyamMattoo** [comment]
@josephjclark Thank you for the detailed overview of the issue and sorry for the previous misconception. After careful consideration and thorough testing, I propose a solution to address the problem at hand.

To ensure seamless execution of the CLI metadata command and avoid potential failures, I suggest implementing an automatic installation mechanism for the required adaptor. This approach aims to streamline user experience by validating the presence of the adaptor and installing it, if necessary, thereby preempting any errors during command execution.

The implementation entails developing a function responsible for adaptor validation and installation utilizing the existing install function from the repo/handler module. By integrating this functionality, users can rely on the metadata command without concerns about manually installing the adaptor beforehand.
![Screenshot (64)](https://github.com/OpenFn/kit/assets/96661612/f3989161-6ea7-4d15-af47-08ce4022bde7)

Regarding testing, I have crafted comprehensive unit tests to validate the adaptor validator and installation functions. Additionally, I've ensured compatibility with existing integration tests for the metadata command.

I believe this approach aligns well with our objectives and enhances the overall usability of the CLI metadata command. Your feedback and suggestions on this proposed solution are highly appreciated.
![Screenshot (63)](https://github.com/OpenFn/kit/assets/96661612/defd866d-6f9b-4469-9e90-2a3d93729524)

Thank you!
